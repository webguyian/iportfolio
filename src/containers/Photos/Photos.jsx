import React from 'react';
import { object } from 'prop-types';

import TabNavigation from 'components/TabNavigation/TabNavigation';
import PhotoGallery from './PhotoGallery/PhotoGallery';

import { useInitialRoute } from 'modules/browser/hooks';

export const FavoritesGallery = props => (
  <PhotoGallery {...props} favoritesOnly />
);

const Photos = props => {
  const { location, match } = props;
  const galleryPath = `${match.path}/gallery`;
  const favoritesPath = `${match.path}/favorites`;
  const isMainRoute = [galleryPath, favoritesPath].includes(location.pathname);
  const tabs = [
    {
      path: galleryPath,
      icon: 'images',
      label: 'Photos',
      component: PhotoGallery
    },
    {
      path: favoritesPath,
      icon: 'heart',
      label: 'Favorites',
      component: FavoritesGallery
    }
  ];
  const baseClass = 'photos-app';

  useInitialRoute(match, tabs[0].path);

  return (
    <div className={baseClass}>
      <TabNavigation showNav={isMainRoute} tabs={tabs} />
    </div>
  );
};

Photos.propTypes = {
  location: object.isRequired,
  match: object.isRequired
};

export default Photos;
