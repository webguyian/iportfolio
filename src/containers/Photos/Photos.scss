.photos-app {
  @include stretch-column-nowrap;

  .ui-tab-navigation-link--active {
    > .ui-icon,
    > .ui-tab-navigation-link-label {
      color: $blue;
    }
  }
}
