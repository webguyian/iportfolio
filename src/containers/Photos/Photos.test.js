import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';

import Photos, { FavoritesGallery } from './Photos';

describe('<Photos />', () => {
  const props = {
    location: {
      pathname: '/photos'
    },
    match: {
      path: '/photos'
    }
  };

  it('renders correctly', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Photos {...props} />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders FavoritesGallery', () => {
    const { asFragment } = render(<FavoritesGallery {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
