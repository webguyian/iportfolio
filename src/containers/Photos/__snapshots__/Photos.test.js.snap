// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Photos /> renders FavoritesGallery 1`] = `
<DocumentFragment>
  <div
    class="photo-gallery"
  >
    <h1
      class="ui-text ui-text--display"
    >
      Favorites
    </h1>
    <div
      class="photo-gallery-empty"
    >
      <p
        class="ui-text ui-text--body ui-text--bold photo-gallery-empty-message"
      >
        No favorites found.
      </p>
      <p
        class="ui-text ui-text--body photo-gallery-empty-message"
      >
        Tap the heart to favorite some of your photos.
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Photos /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="photos-app"
  />
</DocumentFragment>
`;
